analytics.js

```diff
@@ -232,6 +279,20 @@ class Analytics {
         }
       }
 
+      // determine the dataPlaneUrl
+      try {
+        this.serverUrl = this.resolveDataPlaneUrl(response.source.dataPlaneUrls);
+      } catch (e) {
+        handleError(e);
+        this.serverUrl = this.serverUrl ? this.serverUrl : DEFAULT_DATAPLANE_URL;
+      }
```

Comment:
We can move the error handling to the function and simplify the logic better. Upon exceptions or invalid input, you can use the default data-plane URL.

Reply:
This way we can also handle cases if say accessing `response.source.dataPlanes` generates some error.